Custom tag

ZuBB opened the pull request:
this will be a PR that holds a code that implements support of custom tags as markers for stripping code

2 @nuzzio: main points of refactoring your code
* make lines fit to 80 chars limit
* replace _"_ with _'_ where its possible
* move _regexp.escape_ outside of export
* write result of replace whenever file content changed or not
* if _dest_ was not provided, use a _src_ filename
  
Please **do not merge this until** we agree on task, tests, docs etc

ZuBB commented:
Hi @nuzzio!

Recently I started to work on new release of my app where I included version this my changes. I made 2 minor changes, but everything else is really OK. Any plans to review/accept this PR?

boonestars commented on tasks/strip_code.js:
```diff
@@ -8,47 +8,241 @@
+            }
+        };
+
+        // iterate over all specified file groups.
```
Would be nice if there were options to ignore specific files as well

ZuBB replied:
Please explain. I am not sure I understand your point.

boonestars commented on Gruntfile.js:
```diff
@@ -44,18 +44,30 @@ module.exports = function(grunt) {
       pattern_options: {
         options: {
-          pattern: / *console\.log\(['"a-z]+\)\n?/g
+          patterns: / *console\.log\(['"a-z]+\)\n?/g
```
seems like this pattern would suit you better
```
/ *console\.log\([\w\S ]+\)\s?/g
```

ZuBB replied:
Yes, your variant is better. Will take case of that in the evening 

// I do not use this option in my project, so I just went with existing code..

ZuBB replied:
Btw, those that regex is only for internal purpose, so changing it does not help too much. but I will change it

boonestars commented on README.md:
```diff
@@ -93,14 +120,22 @@ The following source code exposes the `bar` function to the public API for testi
+        }
+      ]
     },
     src: 'dist/*.js'
```
@ZuBB in reply to your question below/earlier. In the line here, you can only specify directories or files to strip from, which might end up forcing you to list all but a few files if you wanted to strip using a pattern rather than a start and stop block. For example trying to use a regex pattern to remove all uses of "console.log" in all files of project except for files that need to show logs for reporting or debugging purposes.

ZuBB replied:
If you need to strip most of your files but not all of them just create separate targets inside strip_code task. Also you can filter files that will be processed by `filter` prop (http://gruntjs.com/configuring-tasks#files).

This plugin does 1 job --stipping. If you have difficulties with separating which files should be stripped and which should not that this puzzle should be solved at different 'level'.

I hope I understand you example and hope my answer can help you..

nuzzio commented:
Hi @ZuBB,

It has been a long time, I apologize for that. I just wanted to say thank you very much for the changes that you made. I have incorporated them into the dev branch and pushed them to github. 

I will be finishing up development and testing this weekend and releasing the new version of the task.

Thank you.

ZuBB commented:
Hi @nuzzio!

Thanks for update! I have updated MR accordingly to recent changes in master and added stub file for travis. Now seems everything is 'green' and you should be able to do a merge.

just a question of interest: are you have access to publish new version to npm?

nuzzio commented:
Heya, ZuBB I think you are still using the old branch to add changes. Would you be able to make your changes against the current develop branch? A lot has changed.

I am also in the process of implementing the following requirements:
- Any block must not contain any other block, block start or block end within its space.
- Any start block must only be followed by its own end block.
- If any blocks follow an end block it must be a start block.

ZuBB commented:
That's had been done in my MR ~half a year ago
On Sep 13, 2015 5:08 PM, "Rene Cabral" <notifications@github.com> wrote:

> Heya, ZuBB I think you are still using the old branch to add changes.
> Would you be able to make your changes against the current develop branch?
> A lot has changed.
>
> I am also in the process of implementing the following requirements:
>
>    - Any block must not contain any other block, block start or block end
>    within its space.
>    - Any start block must only be followed by its own end block.
>    - If any blocks follow an end block it must be a start block.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nuzzio/grunt-strip-code/pull/6#issuecomment-139879430>
> .
>

ZuBB commented:
.. All of changes you mentioned. Please check my MR attentively
On Sep 13, 2015 5:29 PM, zvasylvv@gmail.com wrote:

That's had been done in my MR ~half a year ago
On Sep 13, 2015 5:08 PM, "Rene Cabral" <notifications@github.com> wrote:

Heya, ZuBB I think you are still using the old branch to add changes. Would
you be able to make your changes against the current develop branch? A lot
has changed.

I am also in the process of implementing the following requirements:

   - Any block must not contain any other block, block start or block end
   within its space.
   - Any start block must only be followed by its own end block.
   - If any blocks follow an end block it must be a start block.

—
Reply to this email directly or view it on GitHub
<https://github.com/nuzzio/grunt-strip-code/pull/6#issuecomment-139879430>.

nuzzio commented:
Hmm those requirements failed in my tests after  implemented your changes. I added my changes afterwards. I will make formal tests and upload them.

ZuBB commented:
Will look on that tomorrow evening when I back to the city
On Sep 13, 2015 5:44 PM, "Rene Cabral" <notifications@github.com> wrote:

> Hmm those requirements failed in my tests after implemented your changes.
> I added my changes afterwards. I will make formal tests and upload them.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nuzzio/grunt-strip-code/pull/6#issuecomment-139882878>
> .
>

nuzzio commented:
Oh and don't mind the comments about resubmitting the MR, I was mistakenly thinking that you made a new MR @.@;

nuzzio commented:
I see now the actual changes are the minimal, jshint and .travis. Adding those in. Don't worry about the comments regarding resubmitting the MR.
